snuba/consumer.py

```diff
@@ -110,3 +134,193 @@ def flush_batch(self, batch: Sequence[ProcessedMessage]):
                     )
 
             self.producer.flush()
+
+
+class InsertBatchWriter(ProcessingStep[InsertBatch]):
+    def __init__(self, writer: BatchWriter[WriterTableRow]) -> None:
+        self.__writer = writer
+
+        self.__messages: MutableSequence[Message[InsertBatch]] = []
+
+    def poll(self) -> None:
+        pass
+
+    def submit(self, message: Message[InsertBatch]) -> None:
+        self.__messages.append(message)
+
+    def close(self) -> None:
+        if not self.__messages:
+            return
+
+        self.__writer.write(
+            itertools.chain.from_iterable(
+                message.payload.rows for message in self.__messages
+            )
+        )
```

Comment:
why is this in `close` instead of `join` ?

Reply:
I really don't think it's a good fit for either method right now since it's a synchronous/blocking method in an asynchronous/nonblocking design.

Ideally this would initiate an asynchronous write (or close one in progress if we ever switch to a true streaming writer) and then immediately return. I figure we'll want to do this eventually, so I left this here. I don't really think that putting a blocking operation in `join` that disrespects the timeout is a particularly good option either.

Reply:
ok